doc/barefoot_dtel/Dtel-SONiC.md

```diff
@@ -0,0 +1,509 @@
+    HMSET ACL_RULE|DTEL_FLOW_WATCHLIST|FW-2 PRIORITY 10 INT_SESSION INT-1 FLOW_OP INT
+    HSET ACL_RULE|DTEL_DROP_WATCHLIST|DW-1 PRIORITY 10 L4_DST_PORT 2000 DROP_REPORT_ENABLE TRUE
+    
+
+## DTel orchagent
+
+This is a new Orch agent added to handle DTel events. These events are demuxed based on the ConfigDB table name.
+
+Events handled by DtelOrch agent:
+
+    * Set event
+    * Delete event
+
+Most events are handled as shown in the sequence diagram below. Ones which are different are depicted in the following figures.
+
+![Control flow](cflow-1.jpg)
+
+__Figure 2: Generic control flow for DTel events__. 
+
+  
+![Control flow](cflow-2.jpg)
+
+__Figure 3: Control flow for DTel events that depend on other Orch agents__. 
+
+
+### Ref-counted DTel objects
+
+These objects cannot be deleted unless their ref-counts are zero.
+
+* INT sessions referenced by INT watchlists
+* Report sessions referenced by DTel events
+
+![Control flow](cflow-3.jpg "Control flow")
+
+__Figure 4: Control flow for DTel events corresponding to ref-counted objects__. 
+
+  
+### State maintained by DTelOrch agent
+
+**INT session hashmap**
+
+* Key: INT session id
+* Value: 
+	* INT session OID
+	* Reference count
+
+**Report session hashmap**
+
+* Key: Report session ID
+* Value: 
+	* Report session OID
+	* Reference count
+
+**Port hashmap (ports with atleast one queue on which reporting in enabled)**
+
+* Key: ifName
+* Value: 
+	* queue list (all queues for a given port on which reporting is enabled)
+		* Key: queue id
+		* Value:
+			* queue report OID
+
+**Event list**
+
+* Key: Event name
+* Value:
+	* Event OID
+		* Report session ID (reference to the Report session)
+		
+## Acl orchagent
+AclOrch agent will be modified to process the new match fields and actions. 
+New sub-classes to AclRule will be added:
+
+* AclRuleDTelDropWatchList
+* AclRuleDTelFlowWatchList
+
+AclRule objects of the above types are created if ACL table corresponding to the configured rule matches one of:
+
+* ACL_TABLE_DTEL_FLOW_WATCHLIST 
+* ACL_TABLE_DTEL_DROP_WATCHLIST
+
+## Minimum configuration
+Following is the minimum configuration for each DTel feature to detect intended events and generate reports:
+
+**DTel general:**
+1. Switch id
+2. Report session with at least source IP, one destination IP and UDP destination port set
+
+**INT specific:**
+1. INT L4 DSCP: value and mask
+2. At least one INT session (with at least max-hop count set)
+3. At least one flow watchlist with flow-op = INT
+4. At least one sink port
+
+**Postcard specific:**
+1. At least one flow watchlist with flow-op = POSTCARD
+
+**Drop report specific:**
+1. At least one drop watchlist with drop report enabled
+
+**Queue report specific:**
+1. Reporting enabled on at least one queue with some threshold (latency or depth) set or tail drop enabled.
```

Comment:
please note that we have both VS testing as well as ACL testbed which covers ACL and everlfow.
you will need to make sure those are not broken plus add more test cases to cover this.

i don't see any reference to what will happen if a user will try to define an ACL which is not supported. 
As there is an ACL HLD already with flows, you should review it and probably update it with all the ACL changes/ additionals. I IMO this HLD should covers details on AC and flowsL. We should have one place and a reference from here.